### PR TITLE
Use NFT gateway on the wallet page

### PR DESCRIPTION
**Pull request by cypt4**
Resolves https://github.com/brave/brave-browser/issues/28407

Now ipfs-like urls are resolved via new BraveWalletIpfsService. Since this is async call some places were rewriten.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet_ui/common/async/lib.ts [hunk 1]
```diff
@@ -1139,3 +1140,56 @@ export function refreshPortfolioFilterOptions () {
...
+
+// Checks whether set of urls have ipfs:// scheme or are gateway-like urls
+export const areSupportedForPinning = async (urls: string[]) => {
+  const results = await Promise.all(urls.flatMap((v) => extractIpfsUrl(stripERC20TokenImageURL(v))))
+  for (const result of results) {
+    if (!(result?.startsWith(IPFS_PROTOCOL) || false)) {
+      return false
+    }
+  }
+  return true
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/async/lib.ts:1152]:
```suggestion
  return results.some(result => result.startsWith(IPFS_PROTOCOL))
```

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Review by yrliou — APPROVED** [review]
core++

**Review by josheleonard — APPROVED** [review]
frontend ++

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.52.17 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | `bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}`
OS | Windows 11 Version 22H2 (Build 22621.1485)


- Verified steps from [brave/brave-browser#28407](https://github.com/brave/brave-browser/issues/28407)
- Verified displayed NFT image's URL is using the custom gateway as the host
image

**Pull request merged**